### PR TITLE
fix Issue 23256 - must supply -mscrtlib manually when compiling for W…

### PR DESCRIPTION
**Pull request by WalterBright**
…indows

Only check if doing a link.

### CODE REVIEW

File: compiler/src/dmd/mars.d [hunk 1]
```diff
@@ -2595,18 +2595,6 @@ private void reconcileCommands(ref Param params, ref Target target)
...
-        {
-            version (Windows)
-            {
-                VSOptions vsopt;
-                vsopt.initialize();
-                driverParams.mscrtlib = vsopt.defaultRuntimeLibrary(target.is64bit).toDString;
-            }
-            else
-                error(Loc.initial, "must supply `-mscrtlib` manually when cross compiling to windows");
-        }
```
Comment (rainers) [inline compiler/src/dmd/mars.d:2609]: I think removing the assignment here is the wrong approach, as `driverParams.mscrtlib` is also used during compilation from `getTargetInfo`. I'd rather just use the default `libcmt` when cross compiling, too.

Comment (kinke) [reply compiler/src/dmd/mars.d:2609]: When cross-compiling, it should IMO clearly default to the bundled MinGW-based libs - `libcmt` is hardly available on non-Windows systems.

Comment (WalterBright) [reply compiler/src/dmd/mars.d:2609]: It's added back in further down.

Comment (rainers) [reply compiler/src/dmd/mars.d:2609]: Indeed, native builds are not affected by the change. Cross compilation now silently accepts no runtime being passed, though. Some functionality requires it during compilation, e.g. C++ interop in `core.stdcpp.xutility`, so using a reasonable default would be better IMO (`msvcrt120` for the mingw based libs is fine, too).

BTW: I don't expect cross-linking to work from within `dmd` as a `wine` wrapper needs to be used to run the linker (maybe `lld` could cross link).

Comment (kinke) [reply compiler/src/dmd/mars.d:2609]: lld can definitely cross-link (using the `lld-link` driver/symlink). So people then only need the MinGW-based libs from the Windows package - well, once the remaining part of https://issues.dlang.org/show_bug.cgi?id=23092 (`-fPIC`) is fixed as well.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**